### PR TITLE
CoDICE: more test refactor

### PR DESCRIPTION
**Pull request by tech3371**
# Change Summary

## Overview

More clean up

### CODE REVIEW

File: imap_processing/tests/codice/test_codice_l1a.py [hunk 1]
```diff
@@ -227,12 +227,8 @@ def test_lo_sw_species(mock_get_file_paths, codice_lut_path):
 
     val_data = load_cdf(val_path)
 
-    sci_input = ScienceInput("imap_codice_l0_lo-sw-species_20250814_v001.pkts")
-    sci_lut_input = AncillaryInput("imap_codice_l1a-sci-lut_20251007_v001.json")
-    dependency = ProcessingInputCollection(sci_input, sci_lut_input)
-
     # Process the input data
-    processed_data = process_l1a(dependency=dependency)[0]
+    processed_data = process_l1a(dependency=ProcessingInputCollection())[0]
```
Comment (greglucas) [inline imap_processing/tests/codice/test_codice_l1a.py:231]: Does this work because you are setting the `side_effect` above to point at the files you need?

Do you want to make dependency optional to `process_l1a()` so you would only need to handle this in one place and not create the empty InputCollections?

**Review by greglucas — APPROVED** [review]

File: imap_processing/tests/codice/test_codice_l1a.py (hunk 1 above)
Comment (tech3371) [reply imap_processing/tests/codice/test_codice_l1a.py:231]: We do need input collection but since we are doing side effect in test, we don't need to pass those objects, just pass an empty collection. That's why I don't want to make it optional.

**Review by lacoak21 — APPROVED** [review]

**Comment by tech3371** [comment]
pre-commit.ci autofix
